### DISCUSSION TITLE
Using Browsershot on a server

### DISCUSSION THREAD

**Discussion by DevelopmentNightmare** [spatie/browsershot#663] [Q&A] [answered]
I'm trying to use Browsershot on a live server and whenever I try to do something with it, I get the following error:

The command "PATH=$PATH:/usr/local/bin:/opt/homebrew/bin NODE_PATH=`npm root -g` node '/home/ploi/<DOMAIN_NAME>/vendor/spatie/browsershot/src/../bin/browser.js' '{"url":"file:\/\/\/tmp\/387480045-0723827001660554323\/index.html","action":"pdf","options":{"args":["--no-sandbox"],"viewport":{"width":800,"height":600},"displayHeaderFooter":false,"waitUntil":"networkidle0","ignoreHttpsErrors":true,"margin":{"top":"10mm","right":"10mm","bottom":"10mm","left":"10mm"},"printBackground":true}}'" failed. Exit Code: 1(General error) Working directory: /home/ploi/<DOMAIN_NAME>/public Output: ================ Error Output: ================ npm WARN config global `--global`, `--local` are deprecated. Use `--location=global` instead. Error: Failed to launch the browser process! /home/ploi/<DOMAIN_NAME>/node_modules/puppeteer/.local-chromium/linux-1022525/chrome-linux/chrome: error while loading shared libraries: libnss3.so: cannot open shared object file: No such file or directory TROUBLESHOOTING: https://github.com/puppeteer/puppeteer/blob/main/docs/troubleshooting.md at onClose (/home/ploi/<DOMAIN_NAME>/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:289:20) at Interface.<anonymous> (/home/ploi/<DOMAIN_NAME>/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserRunner.js:277:24) at Interface.emit (node:events:539:35) at Interface.close (node:readline:586:8) at Socket.onend (node:readline:277:10) at Socket.emit (node:events:539:35) at endReadableNT (node:internal/streams/readable:1345:12) at processTicksAndRejections (node:internal/process/task_queues:83:21)

Seems to me the issue is libnss3.so, but how and where do I fix it?

I ran a `NPM install` and my PHP code is:

```
        $base64 = Browsershot::html($html)
            ->showBackground()
            ->waitUntilNetworkIdle()
            ->ignoreHttpsErrors()
            ->margins($margins, $margins, $margins, $margins)
            ->noSandbox()
            ->base64pdf();

```

It does work on local.

**Comment by mutschler** [comment, marked as answer]
Well you are missing `libnss3` as the error message stated, try installing it?

```
apt-get install libnss3
```

maybe try

```
apt-get install libnspr4 libnss3 libnss3-nssdb libnss3-tools
```

Sidenote: that's more of a `puppeteer` or chrome problem ;)

**Reply by DevelopmentNightmare** [reply]
This seems to fix it:

sudo apt-get install -y nodejs gconf-service libasound2 libatk1.0-0 libc6 libcairo2 libcups2 libdbus-1-3 libexpat1 libfontconfig1 libgbm1 libgcc1 libgconf-2-4 libgdk-pixbuf2.0-0 libglib2.0-0 libgtk-3-0 libnspr4 libpango-1.0-0 libpangocairo-1.0-0 libstdc++6 libx11-6 libx11-xcb1 libxcb1 libxcomposite1 libxcursor1 libxdamage1 libxext6 libxfixes3 libxi6 libxrandr2 libxrender1 libxss1 libxtst6 ca-certificates fonts-liberation libappindicator1 libnss3 lsb-release xdg-utils wget libgbm-dev libxshmfence-dev

**Reply by mutschler** [reply]
Afaik this are all google-chrome dependencies. So downloading and installing chrome manually should also work

**Reply by ets-company** [reply]
@mutschler I am facing the same issue on the windows system can you have any solution to solve it?
![image](https://user-images.githubusercontent.com/77773287/188130408-d89e2902-74e2-4680-b1b7-92c31991cbe3.png)

**Comment by jeegaje** [comment]
i deploy it at shared hosting, so i not have permission to install those dependencies. What i can do?

**Reply by InsafNilam** [reply]
1) Installing Browsershot on Plesk Obsidian and CentOS 7
[LINK HERE](https://craigsimpson.scot/tools/installing-browsershot-on-centos)

2) Fix for ‘Username Is Not in the Sudoers File. This Incident Will Be Reported’
[LINK HERE](https://www.baeldung.com/linux/username-not-in-sudoers-file)
